LDAP-primer.md

```diff
@@ -101,31 +102,37 @@ look like this:
 
     dn: uid=john.doe,ou=People,dc=example,dc=com
     objectClass: top
+    objectClass: person
+    objectClass: organizationalPerson
     objectClass: inetOrgPerson
     uid: john.doe
     cn: John Doe
     sn: Doe
     userPassword: secret
 
-The `dn` attribute will be explained in the next section. Here, just note that
-there are multiple lines starting with "objectClass". That's because certain
-LDAP attributes can be multi-valued, which is represented in LDIF as separate
+The `dn` line will be explained in the next section. Here, just note that
+there are multiple lines starting with "objectClass". That's because most
+LDAP attributes can have multiple values, which is represented in LDIF as separate
 lines with the same attribute name. Attribute values in an entry form a set:
 there must not be any duplicate values for a given attribute.
 
+Attributes cannot exist without values; there's no concept in LDAP of a `null` value.
+
 Attribute names can be more complicated than described here, and more will be
 said in a later section.
 
 ## 2.2. Entry naming and organization
 
 Entries in an LDAP database are logically organized as a tree, called the __directory
-information tree__ (DIT), and every entry has an attribute describing its position
-in the tree. The attribute is named `dn` for "distinguished name", and is unique
-for an entry. Let's repeat the full entry from the previous section and highlight
-the naming attributes:
+information tree__ (DIT), and every entry has a property describing its position
+in the tree. The property is named `dn` for "distinguished name", and is unique
+for an entry. Note that this is *not* one of the entry attributes. Let's repeat the
+full entry from the previous section and highlight the naming attributes:
 
```

Comment:
I know... But the DN might as well be an attribute, since it has the same syntax in LDIF, and is (philosophically speaking) an entry's attribute (i.e., something associated with an entry). In the interest of precision, I'll keep your reformulation, and reformulate it slightly in turn.